libs/zephyr-xpack-internal/src/xpack-extract/runtime-plugin.ts

```diff
@@ -5,70 +5,40 @@ import type {
-  return (_global as any)[globalKey];
+export interface ZephyrRuntimePluginOTAOptions {
+  /** Called when manifest changes are detected */
+  onManifestChange?: (newManifest: ZephyrManifest, oldManifest?: ZephyrManifest) => void;
+  /** Called when manifest fetch fails */
+  onManifestError?: (error: Error) => void;
+  /** Custom manifest URL (defaults to /zephyr-manifest.json) */
+  manifestUrl?: string;
 }
 
-function setGlobalManifestPromise(promise: Promise<ZephyrManifest | undefined>): void {
-  (_global as any)[globalKey] = promise;
+export interface ZephyrRuntimePluginInstance {
+  /** Refresh the manifest and check for changes */
+  refresh: () => Promise<ZephyrManifest | undefined>;
+  /** Get the current cached manifest */
+  getCurrentManifest: () => Promise<ZephyrManifest | undefined>;
 }
 
+// Global object for browser/Node.js compatibility
+const _global = typeof window !== 'undefined' ? window : globalThis;
+
+// Ensure only one fetch is done by the app
 /**
- * Zephyr Runtime Plugin for Module Federation This plugin handles dynamic remote URL
- * resolution at runtime using beforeRequest hook to mutate URLs on the fly
+ * Enhanced Zephyr Runtime Plugin with caching by application_uid and refresh hooks Now
+ * delegates to createZephyrRuntimePluginWithOTA for consistent behavior
  */
-export function createZephyrRuntimePlugin(): FederationRuntimePlugin {
-  let processedRemotes: Record<string, ZephyrDependency> | undefined;
-
-  // Start fetching manifest immediately
-  let zephyrManifestPromise = getGlobalManifestPromise();
-
-  if (!zephyrManifestPromise) {
-    zephyrManifestPromise = fetchZephyrManifest();
-    setGlobalManifestPromise(zephyrManifestPromise);
-  }
-
-  return {
-    name: 'zephyr-runtime-remote-resolver',
-    async beforeRequest(args) {
-      const zephyrManifest = await zephyrManifestPromise;
-
-      if (!processedRemotes) {
-        processedRemotes = identifyRemotes(args, zephyrManifest);
-      }
-
-      // Extract remote name from args.id (format: "remoteName/componentName")
-      const remoteName = args.id.split('/')[0];
-
-      if (!processedRemotes[remoteName]) {
-        return args; // No matching remote found
-      }
-
-      // Get the resolved URL, checking session storage first
-      const resolvedUrl = getResolvedRemoteUrl(processedRemotes[remoteName]);
-
-      const targetRemote = args.options.remotes.find(
-        (remote) =>
-          hasEntry(remote) && (remote.name === remoteName || remote.alias === remoteName)
-      );
-
-      if (!targetRemote) {
-        return args;
-      }
-
-      // Update the remote entry URL
-      targetRemote.entry = resolvedUrl;
-
-      return args;
-    },
-  };
+export function createZephyrRuntimePlugin(
+  options: ZephyrRuntimePluginOTAOptions = {}
+): FederationRuntimePlugin {
+  // Use the enhanced version but only return the plugin for backward compatibility
+  const { plugin } = createZephyrRuntimePluginWithOTA(options);
```

Comment:
Runtime plugin with OTA should not be the default plugin.
It needs to be recognized and applied only for mobile and only if the user explicitly uses it.

can you make that processing separate?